lib/statsig/evaluator.ex

```diff
@@ -422,27 +425,40 @@ defmodule Statsig.Evaluator do
     hash
   end
 
-  defp get_user_id(user, "userID"), do: try_get_with_lower(user, "userID") |> to_string()
+  defp get_user_id(%User{} = user, "userID"), do: to_string(user.user_id)
 
-  defp get_user_id(user, prop),
-    do: try_get_with_lower(Map.get(user, "customIDs", %{}), prop) |> to_string()
+  defp get_user_id(%User{} = user, prop) do
+    (user.custom_ids || [])
+    |> Keyword.get(String.to_atom(prop))
+    |> to_string()
+  end
 
-  # this is kind of messy, but it should work for now
-  defp get_user_field(user, prop) do
-    case try_get_with_lower(user, prop) do
-      nil -> try_get_with_lower(Map.get(user, "custom", %{}), prop)
-      found -> found
-    end
-    |> case do
-      nil -> try_get_with_lower(Map.get(user, "privateAttributes", %{}), prop)
-      found -> found
+  defp get_user_field(%User{} = user, prop) do
+    prop = String.downcase(prop)
+    case prop do
+      "userid" -> user.user_id
+      "email" -> user.email
+      "ip" -> user.ip
+      "useragent" -> user.user_agent
+      "country" -> user.country
+      "locale" -> user.locale
+      "appversion" -> user.app_version
+      _ ->
+        # Check custom fields if not found in main user fields
+        case try_get_with_lower(user.custom || %{}, prop) do
+          nil -> try_get_with_lower(user.private_attributes || %{}, prop)
+          found -> found
+        end
```

Comment:
this is just
```suggestion
        try_get_with_lower(user.custom || %{}, prop) || try_get_with_lower(user.private_attributes || %{}, prop)
```

and suggest making `try_get_with_lower` to be
```elixir
defp try_get_with_lower(%{} = props, key), Map.get(props, key)
defp try_get_with_lower(_, key), do: try_get_with_lower(%{}, key)
```

which simplifies the first call to 
```elixir
try_get_with_lower(user.custom, prop) || try_get_with_lower(user.private_attributes, prop)
```